### PR TITLE
feat: rpc lib integration

### PR DESCRIPTION
**Pull request by andreabadesso**
https://github.com/user-attachments/assets/3150d6db-f80b-4982-a676-0a57d36f1279

### Acceptance Criteria
- We should enable wallet-connect
- We shouldn't initialize wallet-connect if the wallet-service is enabled
- We should be able to handle the `htr_signMessageWithAddress` and `htr_sendNanoContractTx` requests
- We should display modals for user confirmation and send the response back to wallet connect
- We should crash the entire rootSaga if an error happens in a child saga

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: shim.js [hunk 1]
```diff
@@ -26,3 +26,20 @@ if (typeof localStorage !== 'undefined') {
...
+  global.btoa = function (str) {
+    return Buffer.from(str, 'binary').toString('base64');
+  };
+}
+
+if (typeof atob === 'undefined') {
+  global.atob = function (b64Encoded) {
+    return Buffer.from(b64Encoded, 'base64').toString('binary');
+  };
+}
```
Comment (andreabadesso) [inline shim.js:45]: These are needed by wallet-connect

File: src/App.js [hunk 2]
```diff
@@ -733,8 +736,8 @@ const App = () => (
         >
           <NetworkStatusBar />
           <RootStack />
+          <WalletConnectModal />
```
Comment (andreabadesso) [inline src/App.js:739]: So we're able to navigate from inside a modal

File: src/reducers/reducer.init.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { applyMiddleware, createStore } from 'redux';
+import { applyMiddleware, legacy_createStore as createStore } from 'redux';
```
Comment (andreabadesso) [inline src/reducers/reducer.init.js:1]: We need to refactor this in the future, `createStore` is deprecated

File: src/sagas/index.js [hunk 4]
```diff
@@ -15,23 +15,35 @@ import { saga as permissionsSagas } from './permissions';
...
-  tokensSagas,
-  pushNotificationSaga,
-  networkSettingsSagas,
-  errorHandlerSagas,
-  featureToggleSagas,
-  permissionsSagas,
-  walletConnectSagas,
-  nanoContractSagas,
-];
+function* rootSaga() {
```
Comment (andreabadesso) [inline src/sagas/index.js:21]: This could be a PR on its own, the idea here is to re-create the saga if some unhandled error bubble up to the rootSaga

The current behavior is that if any saga crashes, the root saga crashes and nothing works anymore

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js [hunk 5]
```diff
@@ -80,33 +68,12 @@ export const NewNanoContractTransactionModal = () => {
...
       <ModalBase.DiscreteButton
         title={t`Cancel`}
-        onPress={onDismiss}
+        onPress={onModalDismiss}
       />
     </ModalBase>
   );
 };
 
-const WarnDisclaimer = ({ onReadMore }) => (
```
Comment (andreabadesso) [inline src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:89]: Sent to `components/WalletConnect/WarnDisclaimer` to be reused in other modals

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js [hunk 6]
```diff
@@ -5,64 +5,52 @@
...
-  });
-
-  const onDismiss = () => {
-    dispatch(walletWalletReject());
-    dispatch(setNewNanoContractTransaction({ show: false, data: null }));
-  };
+  const onModalDismiss = useCallback(() => {
+    dispatch(walletConnectReject());
+    onDismiss();
+  }, [onDismiss]);
```
Comment (andreabadesso) [inline src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:36]: `onDismiss` comes as a prop, so its reference might change, so we need to pass it as a dep

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js [hunk 7]
```diff
@@ -5,64 +5,52 @@
...
-    navigation.navigate('NewNanoContractTransactionScreen', { ncTxRequest });
+    onDismiss();
+    navigation.navigate('NewNanoContractTransactionScreen', { ncTxRequest: data });
   };
 
+  useEffect(() => {
+    if (isRetrying) {
+      navigatesToNewNanoContractScreen();
+    }
+  }, [isRetrying, navigation]);
```
Comment (andreabadesso) [inline src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:47]: The idea here is to bypass the initial modal when the user is retrying

Without this "redirect", the user would have to press review in this screen again, after pressing "retry":

image

File: src/sagas/index.js (hunk 4 above)
Comment (andreabadesso) [reply src/sagas/index.js:21]: I'm doing this in this PR because things might fail unexpectedly because of wallet-connect, so this prevents it from propagating to other sagas

File: src/screens/PinScreen.js [hunk 8]
```diff
@@ -157,6 +159,10 @@ class PinScreen extends React.Component {
       if (cb) {
         cb(pin);
       }
+
+      if (this.props.route.params.dispatchResponse) {
+        pinScreenResult(pin);
+      }
```
Comment (andreabadesso) [inline src/screens/PinScreen.js:165]: I could have passed a callback in the route params, but I don't think that's ideal as it yields a warning, I created this new param `dispatchResponse` as a new mechanism for getting the result in the sagas

We should refactor this in a future PR to stop passing callbacks as params

File: locale/pt-br/texts.po [hunk 9]
```diff
@@ -1336,85 +1353,222 @@ msgstr ""
...
+msgstr "Carregando..."
+
+#: src/components/NanoContract/NanoContractTransactionHeader.js:107
+#: src/components/WalletConnect/NanoContract/NanoContractExecInfo.js:128
+msgid "Caller"
+msgstr "Caller"
+
+#: src/components/WalletConnect/NanoContract/NanoContractExecInfo.js:147
+msgid "Couldn't determine address, select one"
+msgstr ""
```
Comment (alexruzenhack) [inline locale/pt-br/texts.po:1420]:
```suggestion
msgstr "Não foi possível determinar um endereço, selecione um"
```

File: locale/pt-br/texts.po [hunk 10]
```diff
@@ -1336,85 +1353,222 @@ msgstr ""
...
+
+#: src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js:27
+msgid "Argument List"
+msgstr "List de Argumentos"
```
Comment (alexruzenhack) [inline locale/pt-br/texts.po:1424]:
```suggestion
msgstr "Lista de Argumentos"
```

File: locale/pt-br/texts.po [hunk 11]
```diff
@@ -1336,85 +1353,222 @@ msgstr ""
...
+msgstr ""
+"Cuidado: Há riscos associados a pedidos de assinatura de transação via dApp."
+
+#: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:102
+msgid "Read More."
+msgstr "Ler mais."
+
+#: src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js:157
+msgid "Nano Contract Not Found"
+msgstr "Nano Contract não nncontrado"
```
Comment (alexruzenhack) [inline locale/pt-br/texts.po:1460]:
```suggestion
msgstr "Nano Contract não encontrado"
```

**Review by alexruzenhack — CHANGES_REQUESTED** [review]

**Review by alexruzenhack — APPROVED** [review]
We discussed the error handling on saga initialization and I'm letting at your discretion to create or not a simple implementation to halt a saga restart, or defer it to another PR.

File: package.json [hunk 12]
```diff
@@ -17,26 +17,35 @@
...
+    "@walletconnect/core": "2.12.2",
+    "@walletconnect/react-native-compat": "2.12.2",
+    "@walletconnect/web3wallet": "1.11.2",
     "assert": "2.0.0",
     "buffer": "4.9.2",
     "console-browserify": "1.2.0",
     "crypto-js": "4.1.1",
     "deprecated-react-native-prop-types": "2.3.0",
-    "events": "1.1.1",
+    "ethers": "5.7.2",
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply package.json:45]: This is required by wallet-connect

File: package.json [hunk 13]
```diff
@@ -17,26 +17,35 @@
...
+    "events": "3.3.0",
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply package.json:46]: Same

File: package.json [hunk 14]
```diff
@@ -17,26 +17,35 @@
...
+    "fast-text-encoding": "1.0.6",
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply package.json:47]: Same

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+  const navigatesToNewNanoContractScreen = () => {
+    onDismiss();
+    navigation.navigate('NewNanoContractTransactionScreen', { ncTxRequest: data });
+  };
+
+  useEffect(() => {
+    if (isRetrying) {
+      navigatesToNewNanoContractScreen();
+    }
+  }, [isRetrying, navigation]);
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js]: Done, thanks

File: src/sagas/walletConnect.js [hunk 16]
```diff
@@ -108,8 +123,13 @@ function* isWalletConnectEnabled() {
 
 function* init() {
   try {
+    const walletServiceEnabled = yield call(isWalletServiceEnabled);
     const walletConnectEnabled = yield call(isWalletConnectEnabled);
 
+    if (walletServiceEnabled) {
+      return;
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/sagas/walletConnect.js:130]: Done, thanks

File: src/sagas/index.js [hunk 17]
```diff
@@ -15,23 +15,35 @@ import { saga as permissionsSagas } from './permissions';
...
-  );
+  yield all(sagas.map(([name, saga]) => spawn(function* supervisor() {
+    while (true) {
+      try {
+        logger('rootSaga').debug(`Starting saga: ${name}`);
+        yield call(saga)
+        break
+      } catch (e) {
+        // TODO: We should have a retry strategy, e.g. if the wallet saga restarts
+        // more than 3 times, we should restart the app and yield a fatal exception
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/sagas/index.js]: I added a retry strategy, can you please check?

**Review by alexruzenhack — APPROVED** [review]

File: src/sagas/wallet.js [hunk 18]
```diff
@@ -390,7 +390,7 @@ export function* fetchTokensMetadata(tokens) {
   const tokenMetadatas = {};
   for (const response of responses) {
     if (response.type === types.TOKEN_FETCH_METADATA_FAILED) {
-      log(`Error downloading metadata of token ${response.tokenId}.`);
+      log.error(`Error downloading metadata of token ${response.tokenId}.`);
```
Comment (andreabadesso) [inline src/sagas/wallet.js:393]: This is unrelated to this PR but would cause a crash in the app

File: src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js [hunk 19]
```diff
@@ -79,11 +77,6 @@ export const NanoContractMethodArgs = ({ blueprintId, method, ncArgs }) => {
       return ncArgs.map((arg, idx) => [methodInfo.args[idx].name, arg]);
     }
 
-    // Send this condition to sentry because it should never happen.
-    // Check any change in the lib or in the fullnode that could cause an impact here.
-    const errMsg = 'Error while getting the argument names of public_methods on blueprint';
-    dispatch(onExceptionCaptured(new Error(errMsg), false));
-
```
Comment (andreabadesso) [inline src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js:86]: This can happen if the dApp sends a request for an invalid blueprint id

And we shouldn't handle this in the component

Comment (andreabadesso) [reply src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js:86]: And it was yielding one exception per argument, which is bad

File: .github/workflows/main.yml [hunk 20]
```diff
@@ -18,15 +18,15 @@ jobs:
...
+        node-version: [18.x, 20.x]
     steps:
     - uses: actions/checkout@v2
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
     - name: Install dependencies
-      run: npm install
+      run: npm ci
```
Comment (andreabadesso) [inline .github/workflows/main.yml:29]: This should cause any changes to package-lock.json to fail

File: package.json [hunk 21]
```diff
@@ -17,26 +17,35 @@
     "postinstall": "rn-nodeify --install stream,process,path,events,crypto,console,buffer --hack"
   },
   "dependencies": {
+    "@ethersproject/shims": "5.7.0",
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply package.json]: Done, can you please check?

File: src/constants.js [hunk 22]
```diff
@@ -248,6 +248,7 @@ export const WALLETCONNECT_NEW_NANOCONTRACT_TX_STATUS = {
 };
 
 export const NANOCONTRACT_BLUEPRINTINFO_STATUS = {
+  READY: 'ready',
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/constants.js]: Removed, thanks

File: package-lock.json
[the remark this answers is not in the document]
Comment (andreabadesso) [reply package-lock.json:1]: I'm not sure why this happened, I tried it using nodejs 18 and nodejs 20, and the results are the same

File: locale/pt-br/texts.po [hunk 23]
```diff
@@ -1637,3 +1667,15 @@ msgstr "Confirma que deseja desregistrar este Nano Contract?"
...
+#~ msgstr "Assinar essa mensagem?"
+
+#~ msgid ""
+#~ "By clicking approve, you will sign the requested message using the first "
+#~ "address derived from your root key on the m/44'/280'/0'/0/0 derivation "
+#~ "path."
+#~ msgstr ""
+#~ "Ao clicar em aprovar, você assinará a mensagem solicitada usando o "
+#~ "primeiro endereço derivado da sua chave raiz no caminho de derivação "
+#~ "m/44'/280'/0'/0/0."
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply locale/pt-br/texts.po]: I'm not sure why these were commented instead of removed

These translations were for the old sign message modal (refactored in this PR)

I removed them

File: src/sagas/wallet.js [hunk 24]
```diff
@@ -387,7 +387,6 @@ export function* loadTokens() {
  * So we fetch the tokens metadata and store on redux
  */
 export function* fetchTokensMetadata(tokens) {
-  // No tokens to load, set metadata as loaded
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/sagas/wallet.js:390]: Because:

1) It's not setting "metadata as loaded", it's just returning
2) It's pointless, you can already see that there are no tokens to load as the length is 0

File: src/screens/PinScreen.js [hunk 25]
```diff
@@ -24,6 +24,7 @@ import {
   onStartWalletLock,
   startWalletRequested,
   resetOnLockScreen,
+  pinScreenResult,
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/screens/PinScreen.js]: Fixed, rebase error, thanks

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js [hunk 26]
```diff
@@ -155,6 +157,7 @@ export const NewNanoContractTransactionRequest = ({ ncTxRequest }) => {
 
   const onFeedbackModalDismiss = () => {
     dispatch(setNewNanoContractStatusReady());
+    dispatch(newNanoContactRetryDismiss());
```
Comment (pedroferreira1) [inline src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js:160]: Why do you need to set retry dismiss here?

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js [hunk 27]
```diff
@@ -5,66 +5,54 @@
...
 import { ModalBase } from '../../ModalBase';
-import SimpleButton from '../../SimpleButton';
-import { setNewNanoContractTransaction, walletConnectReject } from '../../../actions';
-import { WALLET_STATUS } from '../../../sagas/wallet';
+import { walletConnectReject } from '../../../actions';
 import { NANO_CONTRACT_INFO_URL } from '../../../constants';
+import { WarnDisclaimer } from '../WarnDisclaimer';
 
-export const NewNanoContractTransactionModal = () => {
+export const NewNanoContractTransactionModal = ({
```
Comment (pedroferreira1) [inline src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:23]: You also have the `baseStyles` here passed as props, don't you need it?

File: src/components/WalletConnect/SignMessageModal.js [hunk 28]
```diff
@@ -7,61 +7,74 @@
...
+    backgroundColor: COLORS.freeze100,
   },
 });
 
 export default ({
-  onAcceptAction,
-  onRejectAction,
   onDismiss,
   data,
-  baseStyles,
```
Comment (pedroferreira1) [inline src/components/WalletConnect/SignMessageModal.js:32]: Why did you remove this? It's still being passed to the component

File: src/components/WalletConnect/WarnDisclaimer.js [hunk 29]
```diff
@@ -0,0 +1,76 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { t } from 'ttag';
+import { StyleSheet, View, Text } from 'react-native';
+import { COLORS } from '../../styles/themes';
+import { CircleInfoIcon } from '../Icons/CircleInfo.icon';
+import SimpleButton from '../SimpleButton';
+
+export const WarnDisclaimer = ({ onReadMore }) => (
```
Comment (pedroferreira1) [inline src/components/WalletConnect/WarnDisclaimer.js:14]: Why do you have the readMore as props? If the component is used only by dApp warning, it should be the same read more link.

File: src/components/WalletConnect/SignMessageRequest.js [hunk 30]
```diff
@@ -0,0 +1,126 @@
...
+  walletConnectAccept,
+  walletConnectReject
+} from '../../actions';
+import { COLORS } from '../../styles/themes';
+import NewHathorButton from '../NewHathorButton';
+import { DappContainer } from './NanoContract/DappContainer';
+import { commonStyles } from './theme';
+import { NanoContractIcon } from '../Icons/NanoContract.icon';
+
+export const SignMessageRequestData = ({ data }) => (
```
Comment (pedroferreira1) [inline src/components/WalletConnect/SignMessageRequest.js:28]: Do you need to export it?

File: src/components/WalletConnect/SignMessageRequest.js [hunk 31]
```diff
@@ -0,0 +1,126 @@
...
+);
+
+export const SignMessageRequest = ({ signMessageRequest }) => {
+  const { dapp, data } = signMessageRequest;
+  const { message, address } = data;
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+
+  const onAcceptSignMessageRequest = () => {
+    const acceptedNc = { address, message };
```
Comment (pedroferreira1) [inline src/components/WalletConnect/SignMessageRequest.js:59]: What's `Nc` in `acceptedNc`?

File: src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js (hunk 19 above)
Comment (andreabadesso) [reply src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js:86]: I just talked to @alexruzenhack and he reminded me that this is also checking if the method exists in the blueprint, not only if the blueprint exists in the fullnode, so we need to also check for that in the lib PR

File: src/sagas/walletConnect.js [hunk 32]
```diff
@@ -243,19 +281,61 @@ export function* onSessionRequest(action) {
     proposer: get(requestSession.peer, 'metadata.name', ''),
     url: get(requestSession.peer, 'metadata.url', ''),
     description: get(requestSession.peer, 'metadata.description', ''),
-    chain: get(requestSession.namespaces, 'hathor.chains[0]', ''),
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/sagas/walletConnect.js:246]: I shouldn't have removed it, fixed

File: enable-walletconnect.patch [hunk 33]
```diff
@@ -1,60 +1,132 @@
...
-     "buffer": "^4.9.2",
-     "console-browserify": "^1.2.0",
-@@ -67,6 +69,7 @@
-     "redux-saga": "^1.2.0",
-     "redux-thunk": "^2.4.1",
-     "stream-browserify": "^1.0.0",
-+    "text-encoding": "^0.7.0",
-     "ttag": "^1.7.24",
-     "unleash-proxy-client": "^1.11.0",
-     "url": "^0.11.0"
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply enable-walletconnect.patch:21]: I think you're confusing the diff with the github diff

This change is changing the last `enable-walletconnect.patch`

File: src/actions.js [hunk 34]
```diff
@@ -1243,6 +1247,20 @@ export const firstAddressFailure = (failurePayload) => ({
...
+ * Signals that the user wants to attempt to retry the nano contract tx send
+ */
+export const newNanoContactRetry = () => ({
+  type: types.WALLETCONNECT_NEW_NANOCONTRACT_RETRY,
+});
+
+/**
+ * Signals that the user doesn't want to retry the new nano contract tx send
+ */
+export const newNanoContactRetryDismiss = () => ({
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/actions.js]: Fixed, thanks

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js (hunk 27 above)
Comment (andreabadesso) [reply src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:23]: No, this modal has been refactored to a new format created by Alex

Removed the baseStyles prop from where it's not being used 5e84654b352ae96fd25e219aed70b5c94b98ec9f


We can remove it altogether when we refactor the Connect modal to the new style

File: src/components/WalletConnect/SignMessageModal.js (hunk 28 above)
Comment (andreabadesso) [reply src/components/WalletConnect/SignMessageModal.js:32]: Not being used, removed it.

We can remove it from WalletConnectModal when we refactor the Connect modal to use the new modal format Alex created

File: src/components/WalletConnect/SignMessageRequest.js (hunk 30 above)
Comment (andreabadesso) [reply src/components/WalletConnect/SignMessageRequest.js:28]: It is being exported

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js (hunk 26 above)
Comment (andreabadesso) [reply src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js]: Refactored it to run this in the unmount of the component

File: src/components/WalletConnect/SignMessageRequest.js (hunk 31 above)
Comment (andreabadesso) [reply src/components/WalletConnect/SignMessageRequest.js]: "**N**o **c**opypaste" which means that I shouldn't have copy pasted it from other component

refactored it to `acceptedReq`

File: src/components/WalletConnect/WarnDisclaimer.js (hunk 29 above)
Comment (andreabadesso) [reply src/components/WalletConnect/WarnDisclaimer.js]: Refactored it, can you check again?

File: src/sagas/walletConnect.js [hunk 35]
```diff
@@ -243,19 +281,61 @@ export function* onSessionRequest(action) {
...
+    const response = yield call(
+      handleRpcRequest,
+      params.request,
+      wallet,
+      data,
+      promptHandler(dispatch),
+    );
+
+    switch (response.type) {
+      case RpcResponseTypes.SendNanoContractTxResponse:
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/sagas/walletConnect.js:302]: No need because we don't have a "sign success" screen

File: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js [hunk 36]
```diff
@@ -82,33 +70,12 @@ export const NewNanoContractTransactionModal = () => {
...
-          containerStyle={styles.learnMoreContainer}
-          textStyle={styles.learnMoreText}
-          title={t`Read More.`}
-          onPress={onReadMore}
-        />
-      </View>
-    </View>
-  </View>
-);
-
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:111]: You are right, fixed

File: walletconnect.sh [hunk 37]
```diff
@@ -2,4 +2,17 @@
 
 git apply enable-walletconnect.patch
 
+npm install \
```
Comment (andreabadesso) [inline walletconnect.sh:5]: I found it better to just install the dependencies (with fixed versions) than to have them in the patch, as any dependency change would cause the patch to break

File: enable-walletconnect.patch [hunk 38]
```diff
@@ -1,60 +1,199 @@
...
+   React-logger: 2e4aee3e11b3ec4fa6cfd8004610bbb3b8d6cca4
++  react-native-compat: ea766c4bcad72b3cb36765a97a846c9551b5b802
++  react-native-get-random-values: 21325b2244dfa6b58878f51f9aa42821e7ba3d06
++  react-native-netinfo: bdb108d340cdb41875c9ced535977cac6d2ff321
+   react-native-pager-view: da490aa1f902c9a5aeecf0909cc975ad0e92e53e
+   react-native-randombytes: 421f1c7d48c0af8dbcd471b0324393ebf8fe7846
+   react-native-safe-area-context: 68b07eabfb0d14547d36f6929c0e98d818064f02
 diff --git a/shim.js b/shim.js
-index 73c130d..10d894b 100644
+index 73c130d..deae975 100644
```
Comment (pedroferreira1) [inline enable-walletconnect.patch:127]: Why do you have the `shim` file twice here?

File: enable-walletconnect.patch [hunk 39]
```diff
@@ -1,60 +1,199 @@
...
++    return Buffer.from(str, 'binary').toString('base64');
++  };
++}
++
++if (typeof atob === 'undefined') {
++  global.atob = function (b64Encoded) {
++    return Buffer.from(b64Encoded, 'base64').toString('binary');
++  };
++}
 diff --git a/src/sagas/walletConnect.js b/src/sagas/walletConnect.js
```
Comment (pedroferreira1) [inline enable-walletconnect.patch:151]: This file is also duplicated here

File: src/components/WalletConnect/SignMessageRequest.js (hunk 30 above)
Comment (pedroferreira1) [reply src/components/WalletConnect/SignMessageRequest.js:28]: Yes, I know. I just don't see reason for this to be exported but it's ok to keep like this also

File: enable-walletconnect.patch (hunk 38 above)
Comment (andreabadesso) [reply enable-walletconnect.patch:2]: Fixed, can you check again?

File: enable-walletconnect.patch (hunk 39 above)
Comment (andreabadesso) [reply enable-walletconnect.patch:26]: Fixed, can you check again?

**Review by pedroferreira1 — APPROVED** [review]

File: src/components/WalletConnect/SignMessageRequest.js (hunk 30 above)
Comment (andreabadesso) [reply src/components/WalletConnect/SignMessageRequest.js:28]: I think it's nice to export it so we can test it later on
